examples/michael_schumacher_problem/main_michael_schumacher.m

```diff
@@ -136,24 +145,191 @@
     model.lbg_path = -track_width;
     model.ubg_path = +track_width;
 end
+
+
 %% objective
-if ~problem_options.time_optimal_problem 
+if ~problem_options.time_optimal_problem
     model.f_q = u'*u;
-    model.T = 5;
+    problem_options.T = 5;
 else
     model.f_q = 0.01*u'*u;
 end
 
 % Terminal Constraint
-model.g_terminal = q-q_target;
+model.g_terminal = q-q_target; % temrinal constraint expression
 
 % intitial guess for fun
 alpha = [atan2(diff(yy),diff(xx)),0];
-x_guess = vertcat(xx,yy, zeros(2,problem_options.N_stages), alpha);
+x_guess = vertcat(xx,yy, zeros(2,problem_options.N_stages), alpha); % Not used ATM
+
+%% Compute active set guess for mpecopt solver
+active_set_guess = nosnoc.activeset.Pss({[1,2]},'times', [problem_options.T]); % [1,2] is a slinding mode (i.e. c = 0), and it is assumed to hold over [0,T]
+solver_options.mpecopt.initialization_strategy = 'TakeProvidedActiveSet';
+
 %% Solve
 ocp_solver = nosnoc.ocp.Solver(model, problem_options, solver_options);
-ocp_solver.solve();
+ocp_solver.set_initial_active_set(active_set_guess); 
```

Comment:
what active set is taking if this line is not run? 
the first BNLP Thakes then forever. 
Interestingly, with providing the active set, the optimal solution has normal velocity = 0. i.e., it seems no active set changes. with relax and project there are quite a few switches.

Reply:
I don't think it is. the default initialization is the one that is good for Scholtes and co, i,e, _all regions active_.

Reply:
Won't be all zeros but the default `w0` which has all regions active.

Reply:
passing a warning when this is done would be nice.

Reply:
Hmmm not sure where this warning should be checked. I guess in solve? will think about it.